src/Cache.php

```diff
@@ -196,8 +196,9 @@ public function addMultiple(array $values, $ttl = null, Dependency $dependency =
     private function prepareDataForSetOrAddMultiple(iterable $values, ?Dependency $dependency): array
     {
         $data = [];
+        $i = 0;
         foreach ($values as $key => $value) {
-            $value = $this->addEvaluatedDependencyToValue($value, $dependency);
+            $value = $this->addEvaluatedDependencyToValue($value, $dependency, 0 === $i++);
```

Comment:
What this change is for?

Reply:
To make sure that dependency is evaluated once. Because you can pass dependency that was previously evaluated. We need to reevaluate it once for the batch.

Reply:
Looks complicated. The method does two things:

1. Evaluates dependency.
2. Adds it to the value.

If we'll split it into two methods it will be simpler:

```php
$evaluatedDependency = $this->ensureDependencyEvaluated($dependency)
$data = [];
foreach ($values as $key => $value) {
    $data[$this->buildKey($key)] = $this->addDependencyToValue($value, $evaluatedDependency);
}
```